code/game/objects/structures/mystery_box.dm

```diff
@@ -0,0 +1,278 @@
+	/obj/item/gun/ballistic/automatic/pistol/deagle/camo,
+	/obj/item/gun/ballistic/automatic/pistol/suppressed,
+	/obj/item/gun/energy/pulse/carbine,
+	/obj/item/gun/energy/pulse/pistol,
+	/obj/item/gun/ballistic/shotgun/lethal,
+	/obj/item/gun/ballistic/shotgun/automatic/combat,
+	/obj/item/gun/ballistic/shotgun/bulldog,
+	/obj/item/gun/ballistic/rifle/boltaction,
+	/obj/item/gun/ballistic/automatic/ar,
+	/obj/item/gun/ballistic/automatic/proto,
+	/obj/item/gun/ballistic/automatic/c20r,
+	/obj/item/gun/ballistic/automatic/l6_saw,
+	/obj/item/gun/ballistic/automatic/m90,
+	/obj/item/gun/ballistic/automatic/tommygun,
+	/obj/item/gun/ballistic/automatic/wt550,
+	/obj/item/gun/ballistic/automatic/sniper_rifle,
+	/obj/item/gun/ballistic/rifle/boltaction/brand_new,
+))
+
+GLOBAL_LIST_INIT(mystery_box_extended, list(
+	/obj/item/clothing/gloves/tackler/combat,
+	/obj/item/clothing/gloves/race,
+	/obj/item/clothing/gloves/rapid,
+	/obj/item/shield/riot/flash,
+	/obj/item/grenade/stingbang/mega,
+	/obj/item/storage/belt/sabre,
+	/obj/item/knife/combat,
+	/obj/item/melee/baton/security/loaded,
+	/obj/item/reagent_containers/hypospray/combat,
+	/obj/item/defibrillator/compact/combat/loaded/nanotrasen,
+	/obj/item/melee/energy/sword/saber,
+	/obj/item/spear,
+	/obj/item/circular_saw,
+))
+
+
+/obj/structure/mystery_box
+	name = "mystery box"
+	desc = "A wooden crate that seems equally magical and mysterious, capable of granting the user all kinds of different pieces of gear."
+	icon = 'icons/obj/crates.dmi'
+	icon_state = "wooden"
+	pixel_y = -4
+
+	var/crate_open_sound = 'sound/machines/crate_open.ogg'
+	var/crate_close_sound = 'sound/machines/crate_close.ogg'
+	var/open_sound = 'sound/effects/mysterybox/mbox_full.ogg'
+	var/grant_sound = 'sound/effects/mysterybox/mbox_end.ogg'
+	/// The box's current state, and whether it can be interacted with in different ways
+	var/box_state = MYSTERY_BOX_STANDBY
+	/// The object that represents the rapidly changing item that will be granted upon being claimed. Is not, itself, an item.
+	var/obj/mystery_box_item/presented_item
+	/// A timer for how long it takes for the box to start its expire animation
+	var/box_expire_timer
+	/// A timer for how long it takes for the box to close itself
+	var/box_close_timer
+	/// Every type that's a child of this that has an icon, icon_state, and isn't ABSTRACT is fair game. More granularity to come
+	var/selectable_base_type = /obj/item
+	/// The instantiated list that contains all of the valid items that can be chosen from. Generated in [/obj/structure/mystery_box/proc/generate_valid_types]
+	var/list/valid_types
+	/// If the prize is a ballistic gun with an external magazine, should we grant the user a spare mag?
+	var/grant_extra_mag = TRUE
+
+/obj/structure/mystery_box/Initialize(mapload)
+	. = ..()
+	generate_valid_types()
+
+/obj/structure/mystery_box/attack_hand(mob/living/user, list/modifiers)
+	. = ..()
+	switch(box_state)
+		if(MYSTERY_BOX_COOLING_DOWN)
+
```

Comment:
remove empty switch cases, i know why they're there but they don't really help reading that much